src/elexsolver/TransitionSolver.py

```diff
@@ -0,0 +1,111 @@
+import logging
+import warnings
+from abc import ABC
+
+import numpy as np
+
+from elexsolver.logging import initialize_logging
+
+initialize_logging()
+
+LOG = logging.getLogger(__name__)
+
+
+class TransitionSolver(ABC):
```

Comment:
is there any way/ does it make sense for this to inherit from LinearSolver? Then we'd get a lot of the checking functions for free?

Reply:
Interesting idea!  I'll try it!  Thanks! 🤔 🤞🏻

Reply:
Ok, I just pushed changes where `TransitionSolver` inherits from `LinearSolver` 🎉  Curious to know your thoughts 🤔